source/readonly-deep.d.ts

```diff
@@ -62,3 +68,14 @@ Same as `ReadonlyDeep`, but accepts only `object`s as inputs. Internal helper fo
 type ReadonlyObjectDeep<ObjectType extends object> = {
 	readonly [KeyType in keyof ObjectType]: ReadonlyDeep<ObjectType[KeyType]>
 };
+
+/**
+Test if the given function has multiple call signatures. Internal helper for `ReadonlyDeep`.
+Maybe wants to me exported as it's own until type.
+*/
+type HasMultipleCallSignatures<T extends (...arguments: any[]) => unknown> =
+	T extends {(...args: infer A1): any; (...args: any): any}
```

Comment:
Is it possible to use `unknown` for the return values here? I try to use `unknown` whenever possible.